GitUI/UserControls/RevisionGrid/Columns/CommitIdColumnProvider.cs

```diff
@@ -13,6 +13,8 @@ internal sealed class CommitIdColumnProvider : ColumnProvider
     {
         private readonly Dictionary<Font, int[]> _widthByLengthByFont = new Dictionary<Font, int[]>(capacity: 4);
         private readonly RevisionGridControl _grid;
+        public const int DefaultLength = 7;
+        public int Length { get; private set; } = DefaultLength;
```

Comment:
Could you please clarify why this is required?

Reply:
`Length` returns the current length of the commit hashs displayed. It is used by the new context menu item `Copy To Clipboard | Commit hash shortened`.
`DefaultLength` is the fallback for the case that this column has not been shown.
I should have added this as `/// <summary/>`.

Unfortunately, I was not notified by GH on @gerhardol's commits. Thank you for merging!

(Apologies, I beg for patience: Past and this week, I had / will have urgent extensive family commitments, i.e. my time for GE is very limited.)

Reply:
@mstv 
> Unfortunately, I was not notified by GH on @gerhardol's commits. Thank you for merging!

That was in my private test branch that I pushed to test something else, so not merged to master... (This still works for me, no detailed comments on the code.)

Reply:
> Apologies, I beg for patience: Past and this week, I had / will have urgent extensive family commitments, i.e. my time for GE is very limited

No need to apologies, we are all volunteering here, family and RL commitments take precedence.

Reply:
> `Length` returns the current length of the commit hashs displayed. It is used by the new context menu item `Copy To Clipboard | Commit hash shortened`.
> `DefaultLength` is the fallback for the case that this column has not been shown.

I'm not convinced it is a good approach.
Firstly for the reasons @vbjay mentioned above. Git calculates the min length for each repo. Secondly, basing the length on the width of a UI element is incorrect and it **will** come and bite us, if we proceed with this. If we *really* want to offer this functionality we should run the git command.


Also `public const` should not be taken lightly, as they get inlined at compile time. In this instance this must be a `static readonly`.

Reply:
I think I got your point regarding `public const` vs. DLLs. I'll keep it in mind.

Reply:
Yes, this is one of the nuances a lot of engineers don't understand. 
This becomes extremely important when one develops shared assets.